jsonargparse_tests/test_typehints.py

```diff
@@ -605,6 +609,76 @@ def test_typeddict_with_required_arg(parser):
     ctx.match("Expected a <class 'int'>")
 
 
+@pytest.mark.skipif(not Unpack, reason="Unpack introduced in python 3.11 or backported in typing_extensions")
+def test_unpack_support(parser):
+    assert ActionTypeHint.is_supported_typehint(Unpack[Any])
+
+
+if Unpack:  # and Required and NotRequired
+    MyTestUnpackDict = TypedDict("MyTestUnpackDict", {"a": Required[int], "b": NotRequired[int]}, total=True)
+
+    class UnpackClass:
+
+        def __init__(self, **kwargs: Unpack[MyTestUnpackDict]) -> None:
+            self.a = kwargs["a"]
+            self.b = kwargs.get("b")
+
+    # force fallback on AST resolver
+    UnpackByAssumptionsClass = type(
+        "UnpackByAssumptionsClass",
+        (UnpackClass,),
+        {"__module__": __name__, "__init__": lambda self, **kwargs: UnpackClass.__init__(self, **kwargs)},
+    )
+
+    @dataclass
+    class MyTestUnpackClass:
+
+        test: UnpackClass
+
+    @dataclass
+    class MyTestInheritedUnpackClass:
+
+        test: UnpackByAssumptionsClass  # type: ignore[valid-type]
+
+
+@pytest.mark.skipif(not Unpack, reason="Unpack introduced in python 3.11 or backported in typing_extensions")
+def test_unpack_typeddict(parser):
+    parser.add_argument("--testclass", type=MyTestUnpackClass)
+    test_config = {"test": {"class_path": f"{__name__}.UnpackClass", "init_args": {}}}
+    for init_args in [{"a": 1}, {"a": 2, "b": None}, {"a": 3, "b": 1}]:
+        test_config["test"]["init_args"] = init_args
+        cfg = parser.parse_args([f"--testclass={json.dumps(test_config)}"])
+        assert test_config == namespace_to_dict(cfg["testclass"])
+        # also assert no issues with dumping
+        if test_config["test"]["init_args"].get("b") is None:
+            # parser.dump does not dump null b
+            test_config["test"]["init_args"].pop("b", None)
+        assert json.dumps({"testclass": test_config}).replace(" ", "") == parser.dump(cfg, format="json")
+    for init_args in [{}, {"b": None}, {"b": 1}]:
+        with pytest.raises(ArgumentError):
+            test_config["test"]["init_args"] = init_args
+            parser.parse_args([f"--testclass={json.dumps(test_config)}"])
+
+
+@pytest.mark.skipif(not Unpack, reason="Unpack introduced in python 3.11 or backported in typing_extensions")
+def test_unpack_typeddict_by_assumptions(parser):
```

Comment:
Most likely remove this test.

Reply:
I converted this test into a unit test that verifies that `Unpack[TypedDict]` is correctly inferred when using an untyped kwargs in a derived class. 

I'm not 100% certain if this exercises sufficiently distinct logic than the other unit test, so if you think it is superfluous, I can remove it.

Reply:
That is fine.